helm/templates/deployment.yaml

```diff
@@ -2,35 +2,58 @@ apiVersion: extensions/v1beta1
 kind: Deployment
 metadata:
   labels:
-    app.component: {{ .Values.containers.service.name }}
-  name: {{ .Values.containers.service.name }}
+    app.component: {{ .Values.name }}
+  name: {{ .Values.name }}
 spec:
   replicas: {{ .Values.replicas }}
   minReadySeconds: {{ .Values.minReadySeconds }}
   strategy: {}
   template:
     metadata:
       labels:
-        app.component: {{ .Values.containers.service.name }}
+        app.component: {{ .Values.name }}
```

Comment:
I think this label may have been copied from a starter project too.

Reply:
As above, deployment fails without label identifier.

Reply:
this will work if the markup is changed to be the same structure as the gateway and the user service, i.e.
```
    metadata:
      labels:
        app: {{ .Values.name }}
        environment: {{ .Values.environment }}
```

Reply:
Updated as suggested